scipy/stats/tests/test_sobol_.py

```diff
@@ -0,0 +1,212 @@
+
+    def test_UnscrambledFastForwardAndResetSobol(self):
+        engine_unscrambled_3d = Sobol(3, scramble=False).fast_forward(5)
+        draws = engine_unscrambled_3d.random(5)
+        assert_array_equal(self.draws_unscrambled_3d[5:10, :], draws)
+
+        engine_unscrambled_3d.reset()
+        even_draws = []
+        for i in range(10):
+            if i % 2 == 0:
+                even_draws.append(engine_unscrambled_3d.random())
+            else:
+                engine_unscrambled_3d.fast_forward(1)
+        assert_array_equal(
+            self.draws_unscrambled_3d[[i for i in range(10) if i % 2 == 0]],
+            np.vstack(even_draws),
+        )
+
+    def test_UnscrambledHighDimSobol(self):
+        engine = Sobol(1111, scramble=False)
+        count1 = Counter(engine.random().flatten().tolist())
+        count2 = Counter(engine.random().flatten().tolist())
+        count3 = Counter(engine.random().flatten().tolist())
+        assert_equal(count1, Counter({0.0: 1111}))
+        assert_equal(count2, Counter({0.5: 1111}))
+        assert_equal(count3, Counter({0.25: 557, 0.75: 554}))
+
+    def test_UnscrambledSobolBounds(self):
+        engine = Sobol(1111, scramble=False)
+        draws = engine.random(1024)
+        assert_(np.all(draws >= 0))
+        assert_(np.all(draws <= 1))
+
+    def test_UnscrambledDistributionSobol(self):
+        engine = Sobol(1111, scramble=False)
+        draws = engine.random(1024)
+        assert_array_almost_equal(
+            np.mean(draws, axis=0), np.repeat(0.5, 1111), decimal=2
+        )
+        assert_array_almost_equal(
+            np.percentile(draws, 25, axis=0), np.repeat(0.25, 1111), decimal=2
+        )
+        assert_array_almost_equal(
+            np.percentile(draws, 75, axis=0), np.repeat(0.75, 1111), decimal=2
+        )
+
+    def test_Scrambled1DSobol(self):
+        expected = [
+            0.590297,
+            0.46784395,
+            0.03562005,
+            0.91319746,
+            0.86014303,
+            0.23796839,
+            0.25856809,
+            0.63636296,
+            0.69455189,
+            0.316758
+        ]
+        print(self.draws_scrambled_1d.flatten())
+        assert_equal(self.draws_scrambled_1d.shape[0], 10)
+        assert_equal(self.draws_scrambled_1d.shape[1], 1)
+        assert_array_almost_equal(
+            self.draws_scrambled_1d.flatten(), np.array(expected)
+        )
+
+    def test_Scrambled3DSobol(self):
+        expected_dim3 = [0.56645, 0.19712, 0.79965, 0.43654, 0.0867, 0.70811,
+                         0.295, 0.90994, 0.31903, 0.94863]
+        assert_equal(self.draws_scrambled_3d.shape[0], 10)
+        assert_equal(self.draws_scrambled_3d.shape[1], 3)
+        assert_array_almost_equal(
+            self.draws_scrambled_3d[:, 2], np.array(expected_dim3), decimal=5
+        )
+
+    def test_Scrambled3DAsyncSobol(self):
+        engine_unscrambled_3d = Sobol(3, scramble=False)
```

Comment:
This was probably copy-pasted from `test_Unscrambled3DAsyncSobol`, and it actually wasn't wasn't (I assume it was supposed to be). That's a hint that there should be some code-re-use.  With such a large PR, avoiding duplication will really help.

That, combined with comments about what each tests is looking for, will help reviewers wrap their heads around the test suite as a whole to whether it is reasonably complete.